### PR TITLE
Add configuration prefix for rabbit source and sink

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

**Comment by artembilan** [comment]
Looks like nothing to complain about.
Although I wonder if we are going to fix all those `README.adoc` to reflect all these prefix additions...

CC @sabbyanandan

**Comment by markpollack** [comment]
I'm getting the error

``` org.junit.ComparisonFailure:
Expected :"foo"
Actual   :foo
```

for 
`assertEquals("\"foo\"", new String(received.getBody()));`

in  `ExchangeRoutingKeyAndCustomHeaderTests` and `ExchangeRoutingKeyExpressionsAndCustomHeaderTests`

I don't understand why the quote are were explicit in the first place or how this PR changes the behavior wrt to quotes.

**Comment by artembilan** [comment]
@markpollack , `RabbitSinkConfiguration` has this code:

```
@Value("#{${converterBeanName:null}}")
private MessageConverter messageConverter;
```

According to a new prefix logic that should be fixed with the `rabbit.` one:

```
@Value("#{${rabbit.converterBeanName:null}}")
```

Sorry, have missed that from the first sight.

But doesn't work for case of `jsonConverter()` below...
Maybe `static` will help again? :smile: 

Investigating...

**Comment by artembilan** [comment]
Doh. That must be with the prefix, too:

```
@ConditionalOnProperty(prefix = "rabbit", name = "converterBeanName", havingValue = RabbitSinkProperties.JSON_CONVERTER)
    public Jackson2JsonMessageConverter jsonConverter() {
```

File: rabbit/spring-cloud-starter-stream-sink-rabbit/src/main/java/org/springframework/cloud/stream/app/rabbit/sink/RabbitSinkConfiguration.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public RabbitTemplate rabbitTemplate(ConnectionFactory rabbitConnectionFactory)
 	}
 
 	@Bean
-	@ConditionalOnProperty(name = "converterBeanName", havingValue = RabbitSinkProperties.JSON_CONVERTER)
+	@ConditionalOnProperty(name = "rabbit.converterBeanName", havingValue = RabbitSinkProperties.JSON_CONVERTER)
```
Comment (artembilan) [inline rabbit/spring-cloud-starter-stream-sink-rabbit/src/main/java/org/springframework/cloud/stream/app/rabbit/sink/RabbitSinkConfiguration.java:89]: I'd really prefer `prefix` attribute of this annotation instead.
Not sure if that will work if we present the property like though:

```
rabbit.converter-bean-name
...
RABBIT_CONVERTER_BEAN_NAME
```

I mean that there might was a reason to introduce that `prefix` anyway in Boot.

Comment (markpollack) [inline rabbit/spring-cloud-starter-stream-sink-rabbit/src/main/java/org/springframework/cloud/stream/app/rabbit/sink/RabbitSinkConfiguration.java:89]: This kind of stuff is done all over the place in boot's 'common application properties, e.g `server.compression.excluded-user-agents` so it shouldn't be an issue.  See http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html

**Comment by sobychacko** [comment]
The failing tests are fixed.

**Comment by markpollack** [comment]
merged as 786b2f5e5362bbc098d3f8b2bc1d92dba20fdb62

**Pull request closed without merging**